src/Magnum/Platform/GlfwApplication.cpp

```diff
@@ -178,13 +178,24 @@ void GlfwApplication::staticMouseEvent(GLFWwindow*, int button, int action, int)
     } /* we don't handle GLFW_REPEAT */
 }
 
-void GlfwApplication::staticMouseScrollEvent(GLFWwindow*, double xoffset, double yoffset) {
+void GlfwApplication::staticMouseScrollEvent(GLFWwindow* window, double xoffset, double yoffset) {
     MouseScrollEvent e(Vector2d{xoffset, yoffset});
     _instance->mouseScrollEvent(e);
 
     if(yoffset != 0.0) {
-        MouseEvent e1((yoffset > 0.0) ? MouseEvent::Button::WheelUp : MouseEvent::Button::WheelDown);
-        _instance->mousePressEvent(e);
+        InputEvent::Modifiers mods =
+                (glfwGetKey(window, Int(KeyEvent::Key::LeftShift)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightShift)) == GLFW_PRESS)
+                 ? InputEvent::Modifiers{InputEvent::Modifier::Shift} : InputEvent::Modifiers{};
+        if(glfwGetKey(window, Int(KeyEvent::Key::LeftAlt)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightAlt)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::Alt;
+        }
+        if(glfwGetKey(window, Int(KeyEvent::Key::LeftCtrl)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightCtrl)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::Ctrl;
+        }
+        if(glfwGetKey(window, Int(KeyEvent::Key::RightSuper)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::AltGr;
+        }
+        MouseEvent e1((yoffset > 0.0) ? MouseEvent::Button::WheelUp : MouseEvent::Button::WheelDown, mods);
```

Comment:
To have modifiers with all buttons, this monstrosity is necessary. I propose to deprecate Sdl2Application::MouseEvent::Button::Wheel*, since they are simulated anyway, in the same way in which they are simulated here [see this code](https://github.com/mosra/magnum/blob/master/src/Magnum/Platform/Sdl2Application.cpp#L372)